addons/website_event_questions/models/event_event.py

```diff
@@ -17,21 +17,22 @@ class EventEvent(models.Model):
     """ Override Event model to add optional questions when buying tickets. """
     _inherit = 'event.event'
 
-    question_ids = fields.One2many('event.question', 'event_id', 'Questions', copy=True)
+    question_ids = fields.One2many('event.question', 'event_id', 'Questions', copy=True, compute='_compute_from_event_type', store=True, readonly=False)
     general_question_ids = fields.One2many('event.question', 'event_id', 'General Questions',
                                            domain=[('once_per_order', '=', True)])
     specific_question_ids = fields.One2many('event.question', 'event_id', 'Specific Questions',
                                             domain=[('once_per_order', '=', False)])
 
-    @api.onchange('event_type_id')
-    def _onchange_type(self):
-        super(EventEvent, self)._onchange_type()
-        if self.event_type_id.use_questions and self.event_type_id.question_ids:
-            self.question_ids = [(5, 0, 0)] + [
-                (0, 0, {
-                    'title': question.title,
-                    'sequence': question.sequence,
-                    'once_per_order': question.once_per_order,
-                })
-                for question in self.event_type_id.question_ids
-            ]
+    @api.depends('event_type_id')
+    def _compute_from_event_type(self):
+        super(EventEvent, self)._compute_from_event_type()
+        for record in self:
```

Comment:
Other computes do `for event in self`

It's a detail but I think it's easier to read.